std/format.d

```diff
@@ -4362,14 +4362,44 @@ T unformatValue(T, Range, Char)(ref Range input, ref FormatSpec!Char spec)
     return parse!T(input);
 }
 
+private T rawRead(T, Range)(ref Range input)
+    if (isSomeString!Range || ElementType!(Range).sizeof == 1)
+{
+    union X
+    {
+        ubyte[T.sizeof] raw;
+        T typed;
+    }
+    X x;
+    foreach (i; 0 .. T.sizeof)
+    {
+        static if (isSomeString!Range)
+        {
+            x.raw[i] = input[0];
+            input = input[1 .. $];
+        }
+        else
+        {
+            // TODO: recheck this
+            x.raw[i] = cast(ubyte) input.front;
+            input.popFront();
+        }
+    }
+    return x.typed;
+}
+
 /**
    Reads an integral value and returns it.
  */
 T unformatValue(T, Range, Char)(ref Range input, ref FormatSpec!Char spec)
     if (isInputRange!Range && isIntegral!T && !is(T == enum))
```

Comment:
Sorry but I don't get it. All of the other overload functions for unformat value test that T is of a certain type and that input is an InputRange. Why whould the routine for integrals be different?